src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -1778,88 +1757,64 @@ string[] linkingTargetFields
-                    ""enabled"": true,
-                    ""path"": ""/api""
-                    },
-                  ""graphql"": {
-                       ""enabled"": true,
-                       ""path"": ""/graphql"",
-                       ""allow-introspection"": true
-                    }
-                },
-                ""entities"": {
-                    ""HybridEntity"":{
-                        ""source"": {
-                            ""object"": ""hybridSource"",
-                            ""type"":" + $"\"{sourceType}\"" + @"
-                        },
-                        ""permissions"": [
-                           {
-                            ""role"": ""anonymous"",
-                            ""actions"": [
-                               ""*""
-                            ]
-                           }
-                         ],
-                        ""rest"":{
-                            ""methods"":" + $"{methods}" + @"
-                         }
-                       }
-                    }
-                }";
+            Dictionary<string, Entity> entityMap = new();
+            string entityName = "EntityA";
+            // Sets REST method for the entity
+            Entity entity = new(Source: new("TEST_SOURCE", sourceType, null, null),
+                                 Rest: new(Methods: methods),
+                                 GraphQL: new(entityName, ""),
+                                 Permissions: Array.Empty<EntityPermission>(),
+                                 Relationships: new(),
+                                 Mappings: new());
+            entityMap.Add(entityName, entity);
 
-            RuntimeConfigLoader.TryParseConfig(runtimeConfigString, out RuntimeConfig runtimeConfig);
+            RuntimeConfig runtimeConfig = new(
+                Schema: "UnitTestSchema",
+                DataSource: new DataSource(DatabaseType: DatabaseType.MSSQL, string.Empty, Options: null),
+                Runtime: new(
+                    Rest: new(),
+                    GraphQL: new(),
+                    Host: new(null, null)),
+                Entities: new(entityMap));
 
-            // Perform validation on the entity in the config and assert the expected results.
-            if (exceptionExpected)
-            {
-                DataApiBuilderException ex =
-                    Assert.ThrowsException<DataApiBuilderException>(() => RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig));
-                Assert.AreEqual(expectedErrorMessage, ex.Message);
-                Assert.AreEqual(HttpStatusCode.ServiceUnavailable, ex.StatusCode);
-                Assert.AreEqual(DataApiBuilderException.SubStatusCodes.ConfigValidationError, ex.SubStatusCode);
-            }
-            else
+            MockFileSystem fileSystem = new();
+            FileSystemRuntimeConfigLoader loader = new(fileSystem);
+            RuntimeConfigProvider provider = new(loader);
```

Comment:
nit: use the newly added helper IntiatilizeRuntimeConfigObject private function here?

Reply:
Since, the log message is being validated - both the log level and the message logged, we need the instance of the mock logger object. The helper method directly returns the `RuntimeConfigProvider` (which is what is needed in all the other tests).  So, I've not used the helper method for this test.